### PR TITLE
LTD-3758 Type ahead changed to custom sort/filter

### PR DESCRIPTION
**Pull request by Tllew**
### Aim
Have the tokenfield sort by startswith without excluding things that don't start with

[LTD-3768](https://uktrade.atlassian.net/browse/LTD-3758)

[LTD-3768]: https://uktrade.atlassian.net/browse/LTD-3768?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/assets/javascripts/lite-tokenfield-starts-with.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        const bStarts = bName.toLowerCase().startsWith(val);
+        if (aStarts && bStarts) return aName.localeCompare(bName);
+        if (aStarts && !bStarts) return -1;
+        if (!aStarts && bStarts) return 1;
+        return aName.localeCompare(bName);
+      });
+  };
+}
+
+export default window.TokenField = LiteTokenFieldCustomSearch;
```
Comment (kevincarrogan) [inline core/assets/javascripts/lite-tokenfield-starts-with.js:26]: I don't like this side effect here of writing to `window.TokenField`. If we include this version and the other version of token field on the same page then they're going to overwrite each other.

**Review by antroy-madetech — APPROVED** [review]
LGTM

File: caseworker/assets/javascripts/case-notes.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { progressivelyEnhanceMultipleSelectField } from "core/multi-select";
+import { progressivelyEnhanceMultipleSelectField } from "core/multi-select-starts-with";
```
Comment (kevincarrogan) [inline caseworker/assets/javascripts/case-notes.js:1]: I think having this named the same thing as the other one from the other file is going to cause confusion. It won't be obvious that these are different without looking at the imports.

File: core/assets/javascripts/__tests__/multi-select-starts-with.test.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+// make a document with a basic dropdown to instantiate and filter with tokenfield
+
+const createElements = () => {
+  document.body.innerHTML = `
+    <select
+    name="mentions"
+    aria-describedby="id_mentions_hint"
+    class="selectmultiple"
+    id="id_mentions"
+    multiple=""
```
Comment (kevincarrogan) [inline core/assets/javascripts/__tests__/multi-select-starts-with.test.js:11]: We should remove any attributes here that aren't related to the tests themselves.

File: core/assets/javascripts/__tests__/multi-select-starts-with.test.js
Comment (kevincarrogan) [inline core/assets/javascripts/__tests__/multi-select-starts-with.test.js:1]: These tests appear to know a lot about the implementation and are essentially calling a non-public method.

We should write these tests at a higher level to test that the outcome is valid not testing a private method.

File: core/assets/javascripts/multi-select-starts-with.js
Comment (kevincarrogan) [inline core/assets/javascripts/multi-select-starts-with.js:1]: This seems to replicate a lot of what's in the other multiselect file and means these could too easily go out of sync with each other.

File: core/assets/javascripts/__tests__/multi-select-starts-with.test.js
Comment (Tllew) [reply core/assets/javascripts/__tests__/multi-select-starts-with.test.js:1]: I think the name of the file is wrong here, this should be lite-tokenfield-starts-with.test.js

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**